server/src/core/import_resolver.rs

```diff
@@ -185,23 +185,16 @@ pub fn find_module(session: &mut SessionInfo, odoo_addons: Rc<RefCell<Symbol>>,
     let paths = (*odoo_addons).borrow().paths().clone();
     for path in paths.iter() {
         let full_path = Path::new(path.as_str()).join(name.as_str());
-        if is_dir_cs(full_path.sanitize()) {
-            let _arc_symbol = Symbol::create_from_path(session, &full_path, odoo_addons.clone(), false);
-            if _arc_symbol.is_some() {
-                let typ = _arc_symbol.as_ref().unwrap().borrow().typ();
-                match typ {
-                    SymType::NAMESPACE => {
-                        return Some(_arc_symbol.as_ref().unwrap().clone());
-                    },
-                    SymType::PACKAGE(_) => {
-                        let _arc_symbol = _arc_symbol.as_ref().unwrap().clone();
-                        session.sync_odoo.modules.insert(name.clone(), Rc::downgrade(&_arc_symbol));
-                        session.sync_odoo.add_to_rebuild_arch(_arc_symbol.clone());
-                        return Some(_arc_symbol);
-                    },
-                    _ => {return None}
-                }
-            }
+        if !is_dir_cs(full_path.sanitize()) {
+            continue;
+        }
+        let Some(module_symbol) = Symbol::create_from_path(session, &full_path, odoo_addons.clone(), true) else {
+            continue;
+        };
+        if matches!(module_symbol.borrow().typ(), SymType::PACKAGE(PackageType::MODULE)) {
+            session.sync_odoo.modules.insert(name.clone(), Rc::downgrade(&module_symbol));
```

Comment:
I don't think this addition to sync_doo.modules is necessary as create_from_path is already doing it?

Reply:
it is redundant indeed

Reply:
removed